addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/launch-product.test.js

```diff
@@ -0,0 +1,124 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const { iteratee } = require('lodash');
+const WorkflowPayload = require('@aws-ee/workflow-engine/lib/workflow-payload');
+const LaunchProduct = require('../launch-product/launch-product');
+
+describe('LaunchProduct', ()=> {
+      const meta = {};
+      const input = {};
+    
+      const lp = new LaunchProduct({
+        step: { config: {} },
+        workflowPayload: new WorkflowPayload({ meta, input, workflowInstance: { steps: [] } }),
+      });
+
+    // afterEach(() => {
+    //     const resolvedInputParams = [];
+    //     const datetime = 0;
+    //     const datetime2 = 0;
+    // })
+
+    describe('checkNamespace', () => {
+        it('Static name should be transformed to start with analysis- and end with number string and be unique between calls', async () => {
```

Comment:
Nice work catching all the test cases here!

Reply:
Thanks!